enhancements/update/eus-upgrades-mvp.md

```diff
@@ -0,0 +1,368 @@
+current version to the latest .z in the selected channel assuming such a path exists.
+If someone were to provide cluster version 4.4.3 but select the stable-4.6 channel
+no such path exists and it should indicate that, if they were to select stable-4.5
+it would provide the upgrade path of 4.4.3 to 4.4.29 to 4.5.24 which is the latest 4.5.z.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The main caveat of what's outlined here is that it does not reduce reboots
+which, for many reasons, is highly desired by some admins. The rationale for not
+addressing that in this enhancement is that this enhancement targets
+improvements which benefit all upgrades.
+
+The epic / user story to backport API removal notifications to 4.6 assumes that
+we will require anyone upgrading along an EUS specific upgrade path to update to
+a 4.6 version which is specific to EUS. This is necessary because only those who
+follow the EUS upgrade path have 4.10 as their next logical target version. We
+should not alert non EUS clusters of removals in 4.10 until they've upgraded to
+a later minor version.
+
+As with all upgrades, new features enabled by interim versions are not expected
+enabled during the upgrade path. This means that for instance new MachineConfig
+features enabling complex root devices while introduced in 4.7 may not be utilized
+until the cluster has completed its upgrade.
+
+### Risks and Mitigations
+
+The improvements here do not seem to introduce risk unto themselves. However
+failure to implement some of these such as the version skew constraints would
+impose significant risk to supportability in clusters which may upgrade more
+rapidly than is safe to do so.
+
+Additionally, the timeframe for final delivery extends three releases into the
+future. This allows for design iteration during the 4.8 release cycle with
+implementation focused on the 4.9 and 4.10 release cycles.
+
+## Design Details
+
+### Open Questions [optional]
+
+- As this is a bit of a meta-enhancement calling out epic level work it should be
+no surprise that there will be implementation details unaccounted for today. We
+will need to decide on an epic by epic level which require enhancements of their
+own.
+
+- While I believe all of this work to be prerequisite to taking on work to skip
+upgrade steps that may not be widely agreed upon.
+
+- Do we need to consider CCO for similar treatment that we're providing for API
+compatibility? This seems like a nice to have that we could consider for followup.
+If we remova an API workloads break, if credential requests are not capable of
+being met upgrades to a subsequent release will not be possible but the workload
+should remain unaffected.
+
+- Should the graph solver attempt to assist with mirroring of catalog content or
+only core platform content?
+
+- Should we externalize the API removal bits? Perhaps via a marketplace operator
+that's broadly scoped as "EUS 4.6 to EUS 4.10 Validator"?
```

Comment:
If we've written our operators correctly, an EUS Validator would be entirely redundant. All operators must be able to reason about (and take action against!) the validity of their operands in the context of the larger system. This is what makes them an operator as opposed to a controller.

Reply:
I think the idea here is to inform about user workloads wrt deprecated/removed APIs between 4.6 and 4.10, not with our own.

Reply:
Correct, OLM is just the delivery mechanism to install an optional component which could decouple the EUS specific API removal notification from the release payload. I don't think we want to attempt to remediate any issues identified but just inform the admin.